builtin/providers/aws/data_source_aws_eip.go

```diff
@@ -0,0 +1,60 @@
+package aws
+
+import (
+	"fmt"
+	"log"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/ec2"
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+func dataSourceAwsEip() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourceAwsEipRead,
+
+		Schema: map[string]*schema.Schema{
+			"id": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+			},
+
+			"public_ip": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func dataSourceAwsEipRead(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).ec2conn
+
+	req := &ec2.DescribeAddressesInput{}
+
+	if id := d.Get("id"); id != "" {
+		req.AllocationIds = []*string{aws.String(id.(string))}
+	}
```

Comment:
I think commit 5490cc8ec0be2ff0243cc6375fa6eab3d5a7d72f took care of this, and made sure the acceptance test would catch it as well.